source/common/pool_impl.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <stack>
+#include <vector>
+
+#include "nighthawk/common/poolable.h"
+
+#include "common/common/assert.h"
+
+namespace Nighthawk {
+
+template <typename Poolable> class PoolImpl {
```

Comment:
It is weird that you have Poolable both as an ABC and a template name. I think it means that no mehtods are actually called on the ABC; the compiler will know about the derived type during template expansion.